extensions/ql-vscode/src/remote-queries/variant-analysis-manager.ts

```diff
@@ -21,4 +31,51 @@ export class VariantAnalysisManager extends DisposableObject {
   ): Promise<void> {
     await this.variantAnalysisMonitor.monitorVariantAnalysis(variantAnalysis, cancellationToken);
   }
+
+  public async autoDownloadVariantAnalysisResult(
+    scannedRepo: ApiVariantAnalysisScannedRepository,
+    variantAnalysisSummary: VariantAnalysisApiResponse,
+    cancellationToken: CancellationToken
+  ): Promise<void> {
+
+    const credentials = await Credentials.initialize(this.ctx);
+    if (!credentials) { throw Error('Error authenticating with GitHub'); }
+
+    if (cancellationToken && cancellationToken.isCancellationRequested) {
+      return;
+    }
+
+    let repoTask: VariantAnalysisRepoTask;
+    try {
+      repoTask = await ghApiClient.getVariantAnalysisRepo(
+        credentials,
+        variantAnalysisSummary.controller_repo.id,
+        variantAnalysisSummary.id,
+        scannedRepo.repository.id
+      );
+    }
+    catch (e) { throw new Error(`Could not download the results for variant analysis with id: ${variantAnalysisSummary.id}. Error: ${getErrorMessage(e)}`); }
+
+    if (repoTask.artifact_url) {
+      const resultDirectory = path.join(
+        this.ctx.globalStorageUri.fsPath,
+        'variant-analyses',
+        `${variantAnalysisSummary.id}`,
+        scannedRepo.repository.full_name
+      );
+
+      const storagePath = path.join(
+        resultDirectory,
+        scannedRepo.repository.full_name
+      );
+
+      const result = await ghApiClient.getVariantAnalysisRepoResult(
+        credentials,
+        repoTask.artifact_url
+      );
+
+      fs.mkdirSync(resultDirectory, { recursive: true });
+      await fs.writeFile(storagePath, JSON.stringify(result, null, 2), 'utf8');
```

Comment:
What will this actually write to disk? [From the action](https://github.com/github/codeql-variant-analysis-action/blob/0e889fce2f0d6fff82a150d83d36efa12d6b781a/src/query.ts#L170-L187), it seems like we are uploading a ZIP file, so would this write out a binary string encoded as JSON?

Reply:
Thanks! I actually wasn't sure where to check what type of file the result was. 

I don't want this PR to hold up more work and it's going to take a bit more effort to rejig it to handle zip files, so I've written up a separate issue for this.